lib/widgets/swipable_message_row.dart

```diff
@@ -0,0 +1,83 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+
+import '../api/model/model.dart';
+import 'icons.dart';
+import 'text.dart';
+import 'theme.dart';
+
+class SwipableMessageRow extends StatefulWidget {
+  const SwipableMessageRow({
+    super.key,
+    required this.children,
+    required this.message,
+  });
+
+  final List<Widget> children;
+  final Message message;
+
+  @override
+  State<StatefulWidget> createState() => _SwipableMessageRowState();
+}
+
+class _SwipableMessageRowState extends State<SwipableMessageRow> {
+  @override
+  Widget build(BuildContext context) {
+    final hasMarker = widget.message.editState != MessageEditState.none;
+
+    return Row(
+      crossAxisAlignment: CrossAxisAlignment.baseline,
+      textBaseline: localizedTextBaseline(context),
+      children: [
+        hasMarker
+          ? _EditStateMarker(editState: widget.message.editState)
+          : const SizedBox(width: 16),
+        ...widget.children,
+      ],
+    );
+  }
+}
+
+class _EditStateMarker extends StatelessWidget {
+  /// The minimum width of the marker.
+  // Currently, only the collapsed state of the marker has been implemented,
+  // where only the marker icon, not the marker text, is visible.
+  static const double widthCollapsed = 16;
+
+  const _EditStateMarker({
+    required MessageEditState editState,
+  }) : _editState = editState;
+
+  final MessageEditState _editState;
+
+  @override
+  Widget build(BuildContext context) {
+    final designVariables = DesignVariables.of(context);
+
+    final IconData icon;
+    final double iconSize;
+
+    switch (_editState) {
+      case MessageEditState.none:
+        return const SizedBox(width: widthCollapsed);
+      case MessageEditState.edited:
+        icon = ZulipIcons.edited;
+        iconSize = 14;
+        break;
+      case MessageEditState.moved:
+        icon = ZulipIcons.message_moved;
+        iconSize = 8;
```

Comment:
It looks like the underlying icon SVG has a height of 8 but a width of 9. Will this cause it to be scaled down to fit into an 8x8 box?

Reply:
Yeah, that should be the case.

| | moved (8x8) | edited (8x14) |
| - | - | - |
| Flutter | ![Screenshot from 2024-07-12 21-02-36](https://github.com/user-attachments/assets/50400d8c-1e20-470c-bfa1-c03c57739e6a) | ![Screenshot from 2024-07-12 21-01-17](https://github.com/user-attachments/assets/35b332ec-77b0-460a-9f5a-7073df90cbc0) |
| Figma | ![Screenshot from 2024-07-12 21-07-57](https://github.com/user-attachments/assets/9a2bb83a-46ba-4bb9-a97f-093230b8c753) | ![Screenshot from 2024-07-12 21-08-10](https://github.com/user-attachments/assets/01f9a807-6190-4a3c-8ef9-f95e9ff8af1d) |

`iconSize` scales both the width and height of the icon, until it exceeds the constraints of the parent.

<details>

![image](https://github.com/user-attachments/assets/30924df4-97fb-45b6-991f-e238e56aa6e9)

</details>

Reply:
Hmm I see - so the icon file has a width of 9 but the design gives it a width of 8. Confusing. Worth clarifying in a comment, at the least.

Probably also good to ask Vlad about it, in a chat thread in... #mobile, I guess, since it's design (even though it's a fine detail of design).

Reply:
I checked again. So it looks like a later edition has an icon of size 8x8 when downloaded. But I'm starting to think if we should use the frame (8x14) containing that icon, not just the icon itself (8x8) for the design. That way the size also matches the "edited" marker icon.